bossutils/test/test_vault.py

```diff
@@ -61,4 +62,13 @@ def test_exception_if_key_dosent_exist(self, mockClient):
         instance.read.return_value = { "data": {} }
         v = Vault(self.cfg)
         with self.assertRaises(Exception):
-            v.read('secrets', 'super')
\ No newline at end of file
+            v.read('secrets', 'super')
+
+    @patch.object(Vault, 'login')
+    def test_retry_logic(self, mockLogin, mockClient):
+        v = Vault(self.cfg)
+        login = mockLogin.return_value
+        login.login.return_value = True
```

Comment:
Not needed. It is also referencing `Vault.login().login()`. The variable `mockLogin` is the `Vault.login` mock. In any case, the `Vault.login()` method doesn't return anything.